os/drivers/audio/alc5658.c

```diff
@@ -729,41 +748,34 @@ static int alc5658_start(FAR struct audio_lowerhalf_s *dev)
 {
 
 	FAR struct alc5658_dev_s *priv = (FAR struct alc5658_dev_s *)dev;
-	int entry;
-	if (priv->running) {
-		return OK;
-	}
-
 	audvdbg(" alc5658_start Entry\n");
-	alc5658_exec_i2c_script(priv, codec_init_inout_script1, sizeof(codec_init_inout_script1) / sizeof(t_codec_init_script_entry));
+	alc5658_takesem(&priv->devsem);
 
-	entry = alc5658_get_sample_rate_script(priv->samprate);
```

Comment:
If we remove code here, can we change sample rate??

Reply:
Just the order is changed. It is needed for setting sample rate.